components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/messaging_reference.adoc

```diff
@@ -0,0 +1,589 @@
+
+== L1 event reference
+
+Events are listed in alphabetical order.
+
+// Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/IStarknetMessagingEvents.sol[`IStarknetMessagingEvents`]
+
+'''
+
+[#ConsumedMessageToL1]
+=== `ConsumedMessageToL1`
+
+[discrete]
+==== Description
+
+This event is emitted when a message from L2 to L1 is consumed.
+
+[discrete]
+==== Event attributes
+
+[horizontal,role=stripes-odd]
+`uint256 indexed _fromAddress_`:: The address of the sender on L2.
+`address indexed _toAddress_`:: The address of the receiver on L1.
+`uint256[] _payload_`:: The payload of the consumed message.
+
+[discrete]
+==== Event definition
+
+* Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/IStarknetMessagingEvents.sol[`IStarknetMessagingEvents`]
+
+* Event: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/IStarknetMessagingEvents.sol#L19[`ConsumedMessageToL1`]
+
+'''
+
+[#ConsumedMessageToL2]
+=== `ConsumedMessageToL2`
+
+[discrete]
+==== Description
+
+This event is emitted when a message from L1 to L2 is consumed.
+
+[discrete]
+==== Event attributes
+
+[horizontal,role=stripes-odd]
+`address indexed _fromAddress_`:: The address of the sender on L1.
+`uint256 indexed _toAddress_`:: The address of the receiver on L2.
+`uint256 indexed _selector_`::
+include::messaging_reference.adoc[tag=uint256_selector]
+`uint256[] _payload_`:: The payload of the consumed message.
+`uint256 _nonce_`:: The nonce of the consumed message.
+
+[discrete]
+==== Event definition
+
+* Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/IStarknetMessagingEvents.sol[`IStarknetMessagingEvents`]
+
+* Event: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/IStarknetMessagingEvents.sol#L26[`ConsumedMessageToL2`]
+
+'''
+
+[#LogMessageToL1]
+=== `LogMessageToL1`
+
+[discrete]
+==== Description
+
+This event is emitted when a message is sent from L2 to L1.
+
+[discrete]
+==== Event attributes
+
+[horizontal,role=stripes-odd]
+`uint256 indexed _fromAddress_`:: The address of the sender on L2.
+`address indexed _toAddress_`:: The address of the receiver on L1.
+`uint256[] _payload_`:: The payload of the message.
+
+[discrete]
+==== Event definition
+
+* Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/IStarknetMessagingEvents.sol[`IStarknetMessagingEvents`]
+
+* Event: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/IStarknetMessagingEvents.sol#L6[`LogMessageToL1`]
+
+'''
+
+[#LogMessageToL2]
+=== `LogMessageToL2`
+
+[discrete]
+==== Description
+
+This event is emitted when a message is sent from L1 to L2.
```

Comment:
@xiaolou86 Please be consistent with the way you refer to these messages. Let's use:

`...a message from L1 to L2` or `...a message from L2 to L1`.

Please check all occurrences and make them consistent.

Reply:
```suggestion
This event is emitted when sending a message from L1 to L2.
```